### DISCUSSION TITLE
How does Pydantic expose the signature of the __init__ to typecheckers?

### DISCUSSION THREAD

**Discussion by quinor** [pydantic/pydantic#9926] [Question] [answered]
If I write a simple pydantic model and ask pyright for its constructor's type
```
class Foo(pydantic.BaseModel):
    x: int

reveal_type(Foo.__init__)
```
I will get the correct answer. How does Pydantic expose that information to typecheckers?

**Comment by uriyyo** [comment, marked as answer]
That's because of `dataclass_transform` 

https://github.com/pydantic/pydantic/blob/fd8508bef6d131c1c5d07cd2dc087fc2bbf9767a/pydantic/_internal/_model_construction.py#L62-L63

You can find more information here - [PEP 681 – Data Class Transforms](https://peps.python.org/pep-0681/)
